wcomponents-core/src/main/java/com/github/bordertech/wcomponents/WMultiFileWidget.java

```diff
@@ -159,6 +165,21 @@ public FileWidgetUpload getFile(final String fileId) {
 		return null;
 	}
 
+	/**
+	 * Retrieves an mime type of the uploaded file's contents.
+	 * This is not the content type passed by the browser.
+	 *
+	 * @param fileId the file id
+	 * @return an file's mime type, or null if there was no file uploaded
+	 * @throws IOException if there is an error obtaining the input stream from the uploaded file.
+	 */
+	public String getMimeType(final String fileId) throws IOException {
+		if (getFile(fileId) != null) {
+			FileUtil.getFileMimeType(getFile(fileId).getFile());
+		}
+		return null;
+	}
```

Comment:
This function **always** returns `null`. It appears to be an error.

Reply:
No, if the widget has a file, then it's mime will be returned, otherwise null.
I had added [additional class level javadoc](https://github.com/BorderTech/wcomponents/blob/4e6e825546737773489ce40ae9f377734e0e50c4/wcomponents-core/src/main/java/com/github/bordertech/wcomponents/WMultiFileWidget.java#L36) indicating the new API, allowing the client to perform custom mime validations

Reply:
Where are the unit tests?